machine.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: LGPL-2.1+
+import argparse
+import pexpect
+from typing import List, Optional
+import sys
+import time
+import subprocess
+import signal
+import contextlib
+
+import os
+import mkosi
+from mkosi.backend import Distribution, MkosiArgs, MkosiException, ARG_DEBUG, tmp_dir
+
+class Machine():
+    def __init__(self):
+        # Remains None until image is built and booted
+        self.serial = None
+        self.args = None
+
+    def __enter__(self):
+        parsed_args = mkosi.parse_args(sys.argv[1:])
```

Comment:
Instead of using sys.argv, let's only allow the user to pass arguments via the constructor.